osc.js

```diff
@@ -92,584 +579,120 @@ instance.prototype.actions = function (system) {
-						{ id: '1', label: 'prevpage' },
-						{ id: '2', label: 'next playlist item' },
-						{ id: '3', label: 'previous playlist item' },
-					],
-					minChoicesForSearch: 0,
-				},
-				{
-					type: 'checkbox',
-					label: 'Execute',
-					id: 'navigate_to_execute',
-					default: true,
-				},
-			],
-		},
-		navigate_to_playlistitem: {
-			label: 'Navigate to a numbered item within the playlist ',
-			options: [
-				{
-					type: 'textwithvariables',
-					label: 'Playlist item number',
-					id: 'navigate_to_playlistitem',
-					tooltip: 'Frist playlist entry is 0!',
-					default: 1,
-					regex: self.REGEX_SIGNED_NUMBER,
-				},
-			],
-		},
-		video_state: {
-			label: 'Change video state',
-			options: [
-				{
-					type: 'dropdown',
-					label: 'State',
-					id: 'video_state',
-					default: '0',
-					tooltip: 'Choose video state',
-					choices: [
-						{ id: '0', label: 'play' },
-						{ id: '1', label: 'pause' },
-						{ id: '2', label: 'stop' },
-					],
-					minChoicesForSearch: 0,
-				},
-			],
-		},
-		video_position: {
-			label: 'Move to video position',
-			options: [
-				{
-					type: 'textwithvariables',
-					label: 'Position in h',
-					id: 'video_position',
-					tooltip: 'Position of video to skip to as hours with . as decimal',
-					default: 1,
-					regex: self.REGEX_SIGNED_FLOAT,
-				},
-			],
-		},
-		livevideo_state: {
-			label: 'Change livevideo state',
-			options: [
-				{
-					type: 'checkbox',
-					label: 'Play',
-					id: 'livevideo_state',
-					default: true,
-				},
-			],
-		},
-		send_int: {
-			label: 'Send integer',
-			options: [
-				{
-					type: 'textwithvariables',
-					label: 'OSC Path',
-					id: 'path',
-					default: '/osc/path',
-				},
-				{
-					type: 'textwithvariables',
-					label: 'Value',
-					id: 'int',
-					default: 1,
-					regex: self.REGEX_SIGNED_NUMBER,
-				},
-			],
-		},
-		send_string: {
-			label: 'Send string',
-			options: [
-				{
-					type: 'textwithvariables',
-					label: 'OSC Path',
-					id: 'path',
-					default: '/osc/path',
-				},
-				{
-					type: 'textwithvariables',
-					label: 'Value',
-					id: 'string',
-					default: 'text',
-				},
-			],
-		},
-	})
-}
+	}
 
-instance.prototype.action = function (action) {
-	var self = this
+	/**
+	 * Method which is used to run feedbacks something based on the actions defined before
+	 * @param event - The companion event which triggered the feedback
+	 */
+	feedback(event) {
+		if (event.type == 'presentation_state') {
+			this.getVariable('presentation_state', (value) => {
+				var_state = value
+			})
+			states = ['black', 'background', 'page', 'logo']
+			if (var_state == states[event.options.presentation_state]) {
+				return true
+			} else {
+				return false
+			}
+		} else {
+			this.debug('feedback event not recognized', event)
+			false
+		}
+		return false
+	}
 
-	var path
-	var args = null
+	/**
+	 * Default variables available for the module
+	 */
+	variables = [
```

Comment:
Setting this as a class variable is atypical and usually just gets passed to `setVariableDefinitions(…)` in a fashion similar to `actions()` and `this.setActions(…)`